commands/build.js

```diff
@@ -1,234 +1,34 @@
-      argv,
-      "#!/system/bin/script/mxs\nimport './lumin/src/main.js';\n"
-    );
-  }
-}
-
-function buildLumin (argv, indexContent) {
-  let packagePath = 'app.package';
-  try {
-    for (let name of fs.readdirSync('.')) {
-      let m = name.match(/([^/]+)\.package$/);
-      if (!m) continue;
-      let [, base] = m;
-      packagePath = `${base}.package`;
-    }
-  } catch (err) {
-    throw err;
-  }
-
-  var device = argv.debug !== false ? 'device' : 'release_device';
-  var buildCommand = `mabu -t ${argv.host ? 'host' : device} ${packagePath}`;
-
-  // create bin/index.js if needed
-  try {
-    fs.mkdirSync('bin');
-  } catch (error) {
-    if (error.code !== 'EEXIST') {
-      throw error;
-    }
-  }
-  fs.writeFileSync('bin/index.js', indexContent, { mode: 0o755 });
-
-  exec('npm run build', (err, stdout, stderr) => {
-    if (err) {
-      process.stdout.write(stdout);
-      process.stderr.write(stderr);
-      throw err;
-    }
-    util.createDigest(argv.debug);
-    if (argv.host && fs.existsSync('app.package')) {
-      let packageFile = fs.openSync('app.package', 'a+');
-      let buffer = fs.readFileSync(packageFile, 'utf8');
-      if (buffer && !buffer.includes('USES = mxs_lumin_runtime')) {
-        fs.appendFileSync(packageFile, '\nUSES = mxs_lumin_runtime\n');
-      }
-    }
-    exec(buildCommand, (err, stdout, stderr) => {
-      if (err) {
-        process.stdout.write(stdout);
-        process.stderr.write(stderr);
-        throw err;
-      }
-
-      let mpkFile;
-      let outLines = stdout.split('\n');
-
-      let theLine = outLines.find(line => line.includes('mpk'));
-      if (theLine !== undefined) {
-        mpkFile = theLine.substring(theLine.indexOf("'") + 1, theLine.lastIndexOf("'"));
-        console.log('built package: ' + mpkFile);
-      } else {
-        theLine = outLines.find(line => line.includes('output in'));
-        console.log(theLine.substring(theLine.indexOf("'"), theLine.lastIndexOf("'") + 1));
-      }
-
-      if (argv.install) {
-        argv.path = mpkFile;
-        util.installPackage(argv);
-      }
-    });
-  });
-}
-
-module.exports = argv => {
-  npmInstallIfNeeded(`${process.cwd()}`, () => {
-    if (argv.target === 'lumin') {
-      if (isMultiplatformStructure()) {
-        npmInstallIfNeeded(`${process.cwd()}/lumin`, () => {
-          buildLuminComponents(argv);
+const buildUtil = require('../lib/buildutils');
+const logger = require('../lib/logger');
+
+module.exports = (argv) => {
+  if (buildUtil.isTargetSpecified(argv)) {
+    buildUtil.npmInstallIfNeeded(`${process.cwd()}`, () => {
+      if (argv.target === 'lumin') {
+        if (buildUtil.isMultiplatformStructure()) {
+          buildUtil.npmInstallIfNeeded(`${process.cwd()}/lumin`, () => {
+            buildUtil.navigateToLuminDirectory();
+            buildUtil.buildLumin(argv, "#!/system/bin/script/mxs\nimport './lumin/src/main.js';\n");
+          });
+        } else {
+          buildUtil.buildLumin(
```

Comment:
You're calling `buildUtil.buildLumin`  with the same arguments here.  I know one is inside a callback for `npmInstallIfNeeded`, I feel like there's some better way to write this.  I'm particularly worried about this because we have a long string literal here that I'd prefer not to duplicate.  Maybe we can have this string be stored in a `const` at the beginning of this file.